modules/rostests/apitests/user32/CharFuncs.c

```diff
@@ -0,0 +1,574 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for Char* functions
+ * COPYRIGHT:   Copyright 2022 Stanislav Motylkov <x86corez@gmail.com>
+ */
+
+#include "precomp.h"
+
+#include <strsafe.h>
+#include <windows.h>
+#include <ntndk.h>
+
+#define TEST_ON_VISTA  (NTDDI_VERSION >= NTDDI_VISTA)
+#define TEST_ACP       1252
+
+typedef enum
+{
+    testLen,
+    testOffs,
+    testBoth,
+} TEST_TYPE;
+
+/* Dynamic allocation tests */
+struct
+{
+    TEST_TYPE testType;
+    LPWSTR lpszStart;
+    LPWSTR lpszCurrent;
+    INT iOffset;
+    INT iResOffset;
+    INT iResOffsetNeg;
+    BOOL relStart;
+    BOOL bWideOnly;
+} TestChP[] = {
```

Comment:
Don't use abbreviation that is hardly understandable.
`s/TestChP/TestCharPrev/`